src/test/java/net/snowflake/ingest/FipsCompliantIngestIT.java

```diff
@@ -0,0 +1,328 @@
+      {
+        HistoryResponse filesHistory = null;
+        while (true)
+        {
+          Thread.sleep(500);
+          HistoryResponse response = manager.getHistory(null, null, beginMark);
+          if (response.getNextBeginMark() != null)
+          {
+            beginMark = response.getNextBeginMark();
+          }
+          if (response != null && response.files != null)
+          {
+            for (HistoryResponse.FileEntry entry : response.files)
+            {
+              //if we have a complete file that we've
+              // loaded with the same name..
+              String filename = entry.getPath();
+              if (entry.getPath() != null && entry.isComplete() &&
+                  filesWatchList.contains(filename))
+              {
+                if (filesHistory == null)
+                {
+                  filesHistory = new HistoryResponse();
+                  filesHistory.setPipe(response.getPipe());
+                }
+                filesHistory.files.add(entry);
+                filesWatchList.remove(filename);
+                //we can return true!
+                if (filesWatchList.isEmpty()) {
+                  return filesHistory;
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+
+    GetHistory historyCaller = new GetHistory(files);
+    //fork off waiting for a load to the service
+    Future<HistoryResponse> result = service.submit(historyCaller);
+
+    HistoryResponse response = result.get(2, TimeUnit.MINUTES);
+    return response;
+  }
+
+  @Before
+  public void beforeAll() throws Exception
+  {
+    BouncyCastleFIPSInit.init();
+
+    final String filesDirectory = "/tmp/data/";
+    final String filesLocationUrl = "file:///tmp/data/";
+    // The first few steps simulate example files that we will write to the
+    // ingest service.
+    //base file name we want to load
+    final String fileWithWrongCSVFormat = "letters.csv";
+    IngestExampleHelper.makeLocalDirectory(filesDirectory);
+    IngestExampleHelper.makeSampleFile(filesDirectory, fileWithWrongCSVFormat);
+    files.add(IngestExampleHelper.createTempCsv(filesDirectory, "sample", 10).
+        getFileName().toString());
+    files.add(IngestExampleHelper.createTempCsv(filesDirectory, "sample", 15).
+        getFileName().toString());
+    files.add(IngestExampleHelper.createTempCsv(filesDirectory, "sample", 20).
+        getFileName().toString());
+    files.add(fileWithWrongCSVFormat);
+    conn = IngestExampleHelper.getConnection(USER, PASSWORD,
+                                             ACCOUNT, HOST, PORT);
+    keypair = generateKeyPairWithBCFipsProvider();
+    manager = new SimpleIngestManager(ACCOUNT, USER,
+                                      FQ_PIPE, keypair, SCHEME,
+                                      HOST, PORT);
+    setup(files, filesLocationUrl);
+  }
+
+  /**
+   * Remove test table and pipe
+   */
+  @After
+  public void afterAll()
```

Comment:
Shouldn't you also be doing whatever opposite of BouncyCastleFIPSInit.init() is?